presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySortFunction.java

```diff
@@ -79,7 +81,12 @@ public int compare(Integer p1, Integer p2)
                 }
 
                 //TODO: This could be quite slow, it should use parametric equals
-                return type.compareTo(block, p1, block, p2);
+                try {
+                    return type.compareTo(block, p1, block, p2);
+                }
+                catch (PrestoException e) {
+                    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e.getMessage());
```

Comment:
We should only rewrite the exception if the error code is `NOT_SUPPORTED`.

Also, we usually wrap the original exception as the cause in the new exception, e.g. 
```
throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Array contains elements not supported to be compared", e);
```